gapic/schema/api.py

```diff
@@ -902,6 +924,70 @@ def _maybe_get_lro(
 
         return lro
 
+    def _maybe_get_extended_lro(
+        self,
+        service_address: metadata.Address,
+        meth_pb: descriptor_pb2.MethodDescriptorProto,
+    ) -> Optional[wrappers.ExtendedOperationInfo]:
+        op_service_name = meth_pb.options.Extensions[ex_ops_pb2.operation_service]
+        if not op_service_name:
+            return None
+
+        # Manual lookups because services and methods haven't been loaded.
+        # Note: this assumes that the operation service lives in the same proto file.
+        # This is a reasonable assumption as of March '22, but it may (unlikely)
+        # change in the future.
+        op_service_pb = next(
+            (s for s in self.file_descriptor.service if s.name == op_service_name),
+            None,
+        )
+        if not op_service_pb:
+            raise ValueError(
+                f"Could not find custom operation service: {op_service_name}"
+            )
+
+        operation_polling_method_pb = next(
+            (
+                m
+                for m in op_service_pb.method
+                if m.options.Extensions[ex_ops_pb2.operation_polling_method]
+            ),
+            None,
+        )
+        if not operation_polling_method_pb:
+            raise ValueError(
+                f"Could not find operation polling method for custom operation service: {op_service_name}"
+            )
+
+        operation_request_key = service_address.resolve(
+            operation_polling_method_pb.input_type.lstrip(".")
+        )
+        operation_request_message = self.api_messages[operation_request_key]
```

Comment:
Are you certain this won't raise no key found exception ?

Reply:
If it does, we have bigger problems, and we should fail anyway; I could see an argument for checking and explicitly raising the exception with more info, though.